### PR TITLE
Add MW65_r2 Series PCB

### PR DESCRIPTION
**Pull request by TW59420**
## Description
MWStudio is a custom keyboard enthusiast
Upload qmk firmware of mw65_r2 series PCB

## Types of Changes
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mwstudio/mw65_r2/readme.md
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/readme.md:3]:
```suggestion
![MW65_R2](https://i.imgur.com/QzECiuJh.jpg)
```

**Review by drashna — CHANGES_REQUESTED** [review]
You also have a merge conflict that needs resolution.

**Comment by TW59420** [comment]
Please grant my request for merger, Thanks! @drashna

**Comment by drashna** [comment]
> Please grant my request for merger, Thanks! @drashna

You have merge conflicts that must be resolved still.  Until that's done, I can't actually merge it.  Also, we generally want 2 approvals before we merge.

**Comment by TW59420** [comment]
> > 请批准我的合并请求，谢谢！@德拉什纳
> 
> 您有必须仍然解决的合并冲突。在完成之前，我实际上无法合并它。此外，我们通常在合并之前需要 2 个批准。

I see,Sorry! is my problem

**Comment by TW59420** [comment]
@fauxpark  @drashna  Can you help me agree to my pull request?

File: keyboards/mwstudio/mw65_r2/config.h
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/config.h:38]: These aren't really needed, if they're the default. 
```suggestion
```

File: keyboards/mwstudio/mw65_r2/config.h
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/config.h:46]: No animation modes?

File: keyboards/mwstudio/mw65_r2/keymaps/via/keymap.c
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/keymaps/via/keymap.c:93]:
```suggestion
    return false;
```

File: keyboards/mwstudio/mw65_r2/mw65_r2.c
Comment (drashna) [inline keyboards/mwstudio/mw65_r2/mw65_r2.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "mw65_r2.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/mw65_r2/rules.mk
Comment (fauxpark) [inline keyboards/mwstudio/mw65_r2/rules.mk:16]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mwstudio/mw65_r2/config.h
Comment (TW59420) [reply keyboards/mwstudio/mw65_r2/config.h:43]: Just need a light band

**Review by drashna — APPROVED** [review]

**Comment by TW59420** [comment]
@fauxpark  Can you agree to my request? plase，because it's been a long time

File: keyboards/mwstudio/mw65_r2/rules.mk
Comment (fauxpark) [inline keyboards/mwstudio/mw65_r2/rules.mk:16]:
```suggestion
NKRO_ENABLE = yes           # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by TW59420** [comment]
@fauxpark i fix it

**Comment by peepeetee** [comment]
Why the PR closure? Do you need language help? I can offer to add you on QICQ

您怎么把PR关了？您需要语言帮助吗？我可以加您QQ

**Pull request closed without merging**
